calcom/cal.com
# fix: 2FA from CAL to GOOGLE

## What does this PR do?

Switching login from CAL credentials to Google having 2FA enabled was failing due to:
1. Password couldn't be empty which is what it was when entering TOTP code in the 2FA form above the user/password form
2. Condition within authorize logic that missed a check if user was indeed using CAL as provider before asking for TOTP code before setting new provider

Also fixes #7891 regarding being able to disable 2FA for converted CAL to Google providers and UI issues.

## Type of change


## How should this be tested?

1. Sign up using a Google email without using "Sign up with Google"
2. Turn on 2FA
3. Logout
4. Sign in using "Sign in with Google"
5. Provide a TOTP code and should work
6. Try to disable 2FA and you will be able to do that

## Mandatory Tasks
